### PR TITLE
chore: Add custom style hook to component generation templates

### PR DESCRIPTION
**Pull request by ling1726**
see title

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/edc52770091144ba9ddb0fd7705d96833fb3b6ff'>edc52770091144ba9ddb0fd7705d96833fb3b6ff</a></sub>
<div id="bundleSizeComment9423"/>

File: tools/workspace-plugin/src/generators/react-component/files/component/__componentName__.tsx__tmpl__ [hunk 1]
```diff
@@ -12,6 +13,9 @@ export const <%= componentName %>: ForwardRefComponent<<%= componentName %>Props
   const state = use<%= componentName %>_unstable(props, ref);
 
   use<%= componentName %>Styles_unstable(state);
+  use<%= componentName %>Styles_unstable(state);
+  // TODO update types in packages/react-components/react-shared-contexts/src/CustomStyleHooksContext/CustomStyleHooksContext.ts
```
Comment (Hotell) [inline tools/workspace-plugin/src/generators/react-component/files/component/__componentName__.tsx__tmpl__:17]: can we provide more docs on what is this todo about ? why would I need to update types in completely different package ?

Comment (ling1726) [reply tools/workspace-plugin/src/generators/react-component/files/component/__componentName__.tsx__tmpl__:17]: What form would you prefer a longer explanation, would it be OK to just have more inline comments?

Comment (Hotell) [reply tools/workspace-plugin/src/generators/react-component/files/component/__componentName__.tsx__tmpl__:17]: can it point to some md file that provides doc  for it ?

Comment (ling1726) [reply tools/workspace-plugin/src/generators/react-component/files/component/__componentName__.tsx__tmpl__:17]: Added a link to the RFC that proposed the custom styling hook solution https://github.com/microsoft/fluentui/blob/master/rfcs/react-components/convergence/custom-styling.md

**Pull request merged**